test_runner/regress/test_evttrig_extension.py

```diff
@@ -0,0 +1,123 @@
+from fixtures.log_helper import log
+from fixtures.paths import BASE_DIR
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from fixtures.neon_fixtures import (
+        NeonEnvBuilder,
+    )
+    from fixtures.pg_version import PgVersion
+
+
+# use neon_env_builder_local fixture to override the default neon_env_builder fixture
+# and use a test-specific pg_install instead of shared one
+@pytest.fixture(scope="function")
+def neon_env_builder_local(
+    neon_env_builder: NeonEnvBuilder,
+    test_output_dir: Path,
+    pg_distrib_dir: Path,
+    pg_version: PgVersion,
+) -> NeonEnvBuilder:
+    test_local_pginstall = test_output_dir / "pg_install"
+    log.info(f"copy {pg_distrib_dir} to {test_local_pginstall}")
+
+    # We can't copy only the version that we are currently testing because other
+    # binaries like the storage controller need specific Postgres versions.
+    shutil.copytree(pg_distrib_dir, test_local_pginstall)
+
+    # Now copy the SQL only extension test_evttrig_extension in the local
+    # pginstall extension directory on-disk
+    test_evttrig_extension_dir = (
+        BASE_DIR / "test_runner" / "regress" / "data" / "test_evttrig_extension"
+    )
+
+    test_local_extension_dir = (
+        test_local_pginstall / f"v{pg_version}" / "share" / "postgresql" / "extension"
+    )
+
+    log.info(f"copy {test_evttrig_extension_dir} to {test_local_extension_dir}")
+
+    for fn in [
+        test_evttrig_extension_dir / "test_evttrig_extension.control",
+        test_evttrig_extension_dir / "test_evttrig_extension--1.0.sql",
+    ]:
+        shutil.copy(fn, test_local_extension_dir)
+
+    neon_env_builder.pg_distrib_dir = test_local_pginstall
+    log.info(f"local neon_env_builder.pg_distrib_dir: {neon_env_builder.pg_distrib_dir}")
+    return neon_env_builder
+
+
+def test_evttrig_extension(neon_env_builder_local: NeonEnvBuilder):
+    """test installing an extension that contains an event trigger"""
+    env = neon_env_builder_local.init_start()
+    env.create_branch("test_extensions")
+
+    endpoint = env.endpoints.create_start("test_extensions")
+    extension = "test_evttrig_extension"
+    database = "test_extensions"
+
+    SQL_DEFAULT_VERSION = (
+        f"SELECT default_version FROM pg_available_extensions WHERE name = '{extension}'"
+    )
+
+    SQL_NAME_VERSION = f"SELECT extname, extversion FROM pg_extension WHERE extname = '{extension}'"
+
+    endpoint.safe_psql(f"CREATE DATABASE {database}")
+
+    # make sure our test extension is available and not already installed
+    with endpoint.connect(dbname=database) as pg_conn:
+        with pg_conn.cursor() as cur:
+            cur.execute(SQL_DEFAULT_VERSION)
+            res = cur.fetchone()
+            assert res is not None
+            version = res[0]
+
+        with pg_conn.cursor() as cur:
+            cur.execute(SQL_NAME_VERSION)
+            res = cur.fetchone()
+            assert not res, f"The '{extension}' extension is installed"
```

Comment:
Not sure this is a valuable test. Seems like we are testing our own testing infrastructure

Reply:
Remove, thanks for the suggestions.